src/EnergyPlus/DataSurfaces.hh

```diff
@@ -122,6 +149,52 @@ namespace DataSurfaces {
         Count // The counter representing the total number of surface class, always stays at the bottom
     };
 
+    enum class WinShadingType : int {
+        INVALID = -1,
+        NoShade = 0,
+        ShadeOff = 1,
+        IntShade = 2,
+        SwitchableGlazing = 3,
+        ExtShade = 4,
+        ExtScreen = 5,
+        IntBlind = 6,
+        ExtBlind = 7,
+        BGShade = 8,
+        BGBlind = 9,
+        IntShadeConditionallyOff = 10,
+        GlassConditionallyLightened = 11,
+        ExtShadeConditionallyOff = 12,
+        IntBlindConditionallyOff = 13,
+        ExtBlindConditionallyOff = 14,
+        BGShadeConditionallyOff = 15,
+        BGBlindConditionallyOff = 16
+    };
+
+    enum class WindowShadingControlType : int{
+        UnControlled = 0,
+        AlwaysOn = 1,
+        AlwaysOff = 2,
+        OnIfScheduled = 3,
+        HiSolar = 4,
+        HiHorzSolar = 5,
+        HiOutAirTemp = 6,
+        HiZoneAirTemp = 7,
+        HiZoneCooling = 8,
+        HiGlare = 9,
+        MeetDaylIlumSetp = 10,
+        OnNightLoOutTemp_OffDay = 11,
+        OnNightLoInTemp_OffDay = 12,
+        OnNightIfHeating_OffDay = 13,
+        OnNightLoOutTemp_OnDayCooling = 14,
+        OnNightIfHeating_OnDayCooling = 15,
+        OffNight_OnDay_HiSolarWindow = 16,
+        OnNight_OnDay_HiSolarWindow = 17,
+        OnHiOutTemp_HiSolarWindow = 18,
+        OnHiOutTemp_HiHorzSolar = 19,
+        OnHiZoneTemp_HiSolarWindow = 20,
+        OnHiZoneTemp_HiHorzSolar = 21
+    };
```

Comment:
Yeah, I mean, this is just way better. :+1:

Reply:
Mmmmmmmmm. Ok. Also, stick a pin in the fact that we want to provide a portable method for using enumerations in EMS and PythonPlugin so that we can do additional cleanup later.